### PR TITLE
Slingshot Switch Samplers Plus Config fixes

### PR DESCRIPTION
**Pull request by oceandlr**
This includes
1) Slingshot switch samplers
2) Fixes to the top configuration for build checks

This is being submitted as one request with two commits since a merge of the two would necessary anyway

### CODE REVIEW

**Comment by tom95858** [comment]
@morrone, please take a look? I really want to get 4.4.2 out the door.

File: configure.ac [hunk 1]
```diff
@@ -429,6 +429,7 @@ AS_IF([test "x$enable_papi" != xno],[
 ])
 AM_CONDITIONAL([HAVE_LIBPAPI], [test "x$HAVE_LIBPAPI" = xyes])
 AM_CONDITIONAL([HAVE_LIBPFM], [test "x$HAVE_LIBPFM" = xyes])
+AM_CONDITIONAL([ENABLE_PAPI], [test "x$enable_papi" != no -a "x$HAVE_LIBPAPI" = xyes -a "x$HAVE_LIBPFM" = xyes])
```
Comment (morrone) [inline configure.ac:432]: The first "no" here should be "xno"

**Comment by morrone** [comment]
Just the one minor issue, other than that it looks good.

File: configure.ac (hunk 1 above)
Comment (oceandlr) [reply configure.ac:432]: In line 432 you mean?

**Pull request merged**